### PR TITLE
Less autocorrect

### PR DESCRIPTION
**Pull request by bquorning**
Follow-up to #1081. Use `expect_correction` more, and use `include_examples 'autocorrect'` less.

---

Before submitting the PR make sure the following are checked:

* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [-] Updated documentation.
* [-] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
* [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

File: spec/rubocop/cop/rspec/factory_bot/create_list_spec.rb [hunk 1]
```diff
@@ -65,105 +77,135 @@
...
-      expect_offense(<<-RUBY)
-       FactoryGirl.create_list(:user, 3) { |user| user.points = rand(1000) }
-                   ^^^^^^^^^^^ Prefer 3.times.
+      expect_offense(<<~RUBY)
+        FactoryGirl.create_list(:user, 3) { |user| user.points = rand(1000) }
+                    ^^^^^^^^^^^ Prefer 3.times.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        3.times { FactoryGirl.create(:user) } { |user| user.points = rand(1000) }
```
Comment (pirj) [inline spec/rubocop/cop/rspec/factory_bot/create_list_spec.rb:201]: Unrelated to this change, but it turns out this is a bad example - models are persisted before they are passed to the block, so the change to `points` won't be persisted. https://github.com/thoughtbot/factory_bot/issues/1444
We eventually may want to adjust this to `build`, `user.update` or add `user.save!`

File: spec/rubocop/cop/rspec/hook_argument_spec.rb [hunk 2]
```diff
@@ -62,6 +49,13 @@
         config.after(:each)  { true }
         ^^^^^^^^^^^^^^^^^^^ Omit the default `:each` argument for RSpec hooks.
       RUBY
+
+      expect_correction(<<-RUBY)
+        before { true }
+        after  { true }
+        around { true }
+        config.after  { true }
```
Comment (pirj) [inline spec/rubocop/cop/rspec/hook_argument_spec.rb:57]: Cosmetic: extra space, in the "bad" example as well.

File: spec/rubocop/cop/rspec/hook_argument_spec.rb [hunk 3]
```diff
@@ -152,20 +168,35 @@
...
+
+      expect_correction(<<-RUBY)
+        before(:example) { true }
+        after(:example)  { true }
+        around(:example) { true }
+        config.before(:example) { true }
+      RUBY
     end
 
-    it 'does not flag hooks without default scopes' do
```
Comment (pirj) [inline spec/rubocop/cop/rspec/hook_argument_spec.rb:157]: 👍

File: spec/rubocop/cop/rspec/hook_argument_spec.rb (hunk 2 above)
Comment (bquorning) [reply spec/rubocop/cop/rspec/hook_argument_spec.rb]: Fixed

**Pull request merged**